packages/xcm-estimate/README.md

```diff
@@ -4,8 +4,91 @@
 
 ## Installation
 
-TBD
+Before installing `xcm-estimate` package to your project, you need to setup Yarn 2, due to requirements of the `open-xcm-tools` packages.
+
+### Yarn 2 (Berry) installation
+
+First, set up a global Yarn, which will be used to create local instances:
+
+```bash
+npm install -g yarn
+```
+
+Then you need to locally change **Yarn** version for your project. In project folder:
+
+```bash
+yarn set version berry
+```
+
+After this actions you can use **Yarn** as usually (init project, add dependencies, etc.)
+
+### `xcm-estimate` installation
+
+```bash
+yarn add @open-xcm-tools/xcm-estimate
+```
 
 ## Usage
 
-TBD
+The `Estimator` class provides a comprehensive framework for estimating fees and execution effects of cross-chain message (XCM) programs using Runtime API. It enables seamless interactions and efficient resource management by leveraging dry-run simulations and dynamic fee calculations based on the connected chain's XCM Runtime API reports.
+
+### Connect to the Blockchain
+
+Define the chain information, including the WebSocket endpoint and the chain identity:
+
+```typescript
+const chainInfo: ChainInfo = {
+  identity: {
+    name: 'AssetHub',
+    universalLocation: parachainUniversalLocation('polkadot', 2001n),
+  },
+  endpoints: [
+    'wss://asset-hub-polkadot-rpc.dwellir.com',
+    'wss://sys.ibp.network/asset-hub-polkadot',
+    // Add more endpoints as needed
+  ],
+};
+```
+
+Use the constructor of `Estimator` to create an instance of this class:
+
+```typescript
+const api = new ApiPromise.create({provider: provider});
+const estimator = new Estimator(api, chainInfo.identity, 4);
+```
+
+> Note: Also `Estimator.connect` method performs the same function, and you may choose to use it as well.
+
+```typescript
+      Estimator.connect(
+        this.registry.chainInfoByUniversalLocation(universalLocation), // pass only the chain info
+      ),
+```
```

Comment:
I will fix this myself, this is a note to myself.

----

### Estimator creation

Define the chain identity:

```typescript
const chainIdentity: ChainIdentity = {
    name: 'AssetHub',
    universalLocation: parachainUniversalLocation('polkadot', 1000n),
};
```

Use the constructor of `Estimator` to create an instance of this class:

```typescript
const api = /* an instance of the `ApiPromise` */;
const xcmVersion = /* xcm version to use to communicate with the chain */
const estimator = new Estimator(api, chainIdentity, xcmVersion);
```

The `Estimator` can estimate the XCM version by itself (see `estimateMaxXcmVersion`) and it also can create the `ApiPromise` instance internally. If you want to pass the `ChainInfo` (which is the chain identity + list of endpoints) and make the Estimator to "just connect", you can use the `connect` method.

```typescript
      Estimator.connect(<ChainInfo>{
         identity: {
            name: 'AssetHub',
            universalLocation: parachainUniversalLocation('polkadot', 1000n),
         },
         endpoints: [
             'wss://asset-hub-polkadot-rpc.dwellir.com',
             'wss://sys.ibp.network/asset-hub-polkadot',
            // Add more endpoints as needed
         ],
     }),
```